### ISSUE TITLE
Intermittent 403 Errors After Cilium Upgrade

### ISSUE THREAD

**Issue by isala404** [cilium/cilium#32829]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

We were running Cilium 1.13.5 in a ~60 node cluster with around 4,500 pods. After upgrading to Cilium 1.14.7, some services started intermittently returning 403s.

Through Hubble inspection, we noticed that when Pod A tried to connect to Pod B, which had a Cilium proxy annotation on its container port to route traffic through the Cilium Envoy for observability, the Envoy set the verdict of that request as denied. Further investigation revealed that the caller Pod A didn't have an identity associated with it.

The connection between Pod A and Pod B was working before, and both had an age of 15+ days. The intermittent failure was caused by Pod A being part of a ReplicaSet with 3 pods, and only a single replica didn't have an identity. Restarting the ReplicaSet resolved the issue.

Initially, we dismissed this as a one-time occurrence during the Cilium upgrade process.

However, a few days later, we encountered a similar issue in another cluster with 5 nodes running Cilium 1.14.7. Our SRE team immediately restarted the problematic workload, and during the post-mortem, we noticed the failed pods had 403 Access Denied errors in their logs. Notably, this occurred around 18 hours after a Kubernetes upgrade, although no Cilium upgrades were performed.

Currently, we are in a critical state, afraid to make any changes in the cluster, as it could result in the failure of critical services without immediate notice.

### Cilium Version

v1.14.7

### Kernel Version

5.15.0-1060-azure

### Kubernetes Version

1.28.5

### Cilium Users Document

- [x] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by ti-mo** [comment]
Hi @isala404, this reads like a support request, not a bug report. Support is not provided on the issue tracker.

Have you been able to investigate more in your dev cluster? The information you've provided is mostly anecdotal. In order for a dev to pick this up and make meaningful progress on this, we'll need as many specifics as possible. A few starting points could be a sysdump of a currently affected cluster or set of nodes, and the source and target version of the k8s upgrade you performed.

**Comment by isala404** [comment]
I apologize for any confusion in my initial comment. We believe this is a bug because Hubble clearly showed verdict denied for requests from one replica, while the other replica's requests were forwarded. The caller pod's identity was set to unknown. Deleting the affected replica instantly resolved the issue, and Hubble traces showed the new pod got the correct identity with requests being forwarded. We're confident this wasn't due to a service or infrastructure issue.

> Have you been able to investigate more in your dev cluster?

Sadly, We can't reproduce the issue as it occurs randomly. It's been over a month without recurrence. We'll capture a sysdump when it happens again.

> source and target version of the k8s upgrade you performed.

The Kubernetes upgrade was from 1.27.15 to 1.28.5, but we don't think it's related. It was mentioned only for context.

We reported this to check if it's a known issue and if not, to share recovery steps for others experiencing similar problems. We'll send sysdump and Hubble traces when the issue recurs.

**Comment by isala404** [comment]
Hi all,

We recently performed a Cilium upgrade to version 1.15.x across all of our clusters, and we encountered the same issue on a few of our AWS clusters this time as well. In AWS, we are running Cilium in ENI mode. Notably, this time around, all three instances faced the issue where our ingress controller was unable to resolve DNS due to the policy being declined. Restarting the ingress controller pod resolved the issue. We would appreciate any insights into this issue.
